src/components/SigningModals/MultiScreenModal.js

```diff
@@ -0,0 +1,180 @@
+import React, { useCallback, useState } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Modal,
+  Viewport,
+  springs,
+  textStyle,
+  useViewport,
+  GU,
+} from '@aragon/ui'
+import { Spring, Transition, animated } from 'react-spring'
+import { useSteps } from '../../hooks'
+
+const AnimatedDiv = animated.div
+
+const DEFAULT_MODAL_WIDTH = 700
+
+function MultiScreenModal({ visible, screens, onClose }) {
+  const steps = screens.length
+  const { prev, step, next, direction } = useSteps(steps)
+
+  const [measuredHeight, setMeasuredHeight] = useState(false)
+  const [height, setHeight] = useState(null)
+  const [firstStart, setFirstStart] = useState(true)
+  const { below } = useViewport()
+
+  const smallMode = below('medium')
+  const { disableClose, width } = screens[step]
+
+  const modalWidth = width || DEFAULT_MODAL_WIDTH
+
+  const renderScreen = useCallback(
+    step => {
+      const { title, content } = screens[step]
+
+      return (
+        <React.Fragment>
+          <h1
+            css={`
+              ${smallMode ? textStyle('title4') : textStyle('title3')};
+
+              margin-top: -${1 * GU}px;
+              margin-bottom: ${2 * GU}px;
+            `}
+          >
+            {title}
+          </h1>
+          {content({
+            prevScreen: prev,
+            nextScreen: next,
+            closeModal: onClose,
+          })}
+        </React.Fragment>
+      )
+    },
+    [prev, next, onClose, screens, smallMode]
+  )
+
+  const onStart = useCallback(() => {
+    // Don’t animate or set the static height when the modal first opens
+    if (firstStart) {
+      setFirstStart(false)
+
+      return
+    }
+
+    setMeasuredHeight(true)
+  }, [firstStart])
+
+  const handleModalClose = useCallback(() => {
+    if (!disableClose) {
+      onClose()
+    }
+  }, [disableClose, onClose])
+
+  return (
+    <Viewport>
+      {({ width: viewportWidth }) => {
+        const gutter = 30
+        const containWidth = viewportWidth < modalWidth + gutter
+
+        return (
+          <Modal
+            padding={0}
+            width={containWidth ? viewportWidth - gutter : modalWidth}
+            onClose={handleModalClose}
+            visible={visible}
+            closeButton={!disableClose}
+          >
+            <Spring
+              config={springs.swift}
+              to={{ height }}
+              immediate={firstStart}
+              native
+            >
+              {({ height }) => (
+                <AnimatedDiv
+                  style={{
+                    position: 'relative',
+                    height: measuredHeight ? height : 'auto',
```

Comment:
It seems a bit weird to have two state variables for this, when it seems like we should be able to reset `height` to `null` when we want `auto`.

But otherwise, we may want to double check that `height` exists so we avoid a `height: null` declaration :).

Reply:
I agree it would be cleaner, but the issue is we need to record the height independently of applying it due to how manipulations are working with Spring their interactions on the very first render after opening a modal. https://github.com/aragon/aragon/blob/232dc29dfdcc186b929ca1f553cba8aba95eea1c/src/components/AccountModule/AccountModulePopover.js#L93

I think renaming the flag would make it clearer though.